qa/sc_evm_forging_fee_payments.py

```diff
@@ -351,12 +380,15 @@ def run_test(self):
         node_2_fees = per_block_fee * 3
 
         sc_node_1_balance_before_payments = sc_node_1_balance_after_payments
-        sc_node_2_balance_before_payments = sc_node_2_balance_after_payments
+        sc_node_2_balance_before_payments = sc_node_2_balance_after_payments + reward_address_balance
         sc_node_1_balance_after_payments = sc_node_1.wallet_getTotalBalance()['result']['balance']
         sc_node_2_balance_after_payments = sc_node_2.wallet_getTotalBalance()['result']['balance']
         assert_equal(sc_node_1_balance_before_payments + node_1_fees, sc_node_1_balance_after_payments,
                      "Wrong fee payment amount for SC node 1")
-        assert_equal(sc_node_2_balance_before_payments + node_2_fees, sc_node_2_balance_after_payments,
+        # SC node 2 is configured to send rewards to some other address
+        reward_address_balance = http_wallet_balance(sc_node_2, self.FORGER_REWARD_ADDRESS)
+        assert_equal(sc_node_2_balance_before_payments + node_2_fees,
+                     sc_node_2_balance_after_payments + reward_address_balance,
```

Comment:
These 3 cases require each a separate test. I tested all these conditions but decided to stop on adding a test that uses as a reward address a smart contract that does not have payable fallback or receive function (2). But let me describe other tests also:

1. Reward address as smart contract with payable fallback or receive function - smart contract receives funds as expected, and balance can be checked from inside a smart contract

2. Reward address as smart contract without payable fallback or receive function - smart contract also receives funds, and balance can be checked from inside a smart contract. I think this is because we use libevm StateAddBalance -> go-ethereum state.AddBalance -> go-ethereum state_object.AddBalance.
None of the methods have a check for fallback or receive function, so it succeeds.

3. Reward address as FORGER_POOL_RECIPIENT_ADDRESS - this helped me stop a bug, thanks for a suggestion. It works fine, after the fee is calculated, forger pool address is set to 0, and then rewards are distributed. So the reward would be accumulated for a next epoch fee distribution.

Test 1 is a copy of test 2 but with a slightly modified Smart Contract used - does not affect test execution.
Test 3 is a copy of sc_evm_forging_fee_payments.py where reward address is set to forger_reward_pool. There is also extra calculation required for a seconds fee distribution, as forger_reward_pool is bigger after it receives funds at the end of first epoch.